File: app/src/main/java/ua/hospes/nfs/marathon/data/race/models/RaceItemDb.java
package ua.hospes.nfs.marathon.data.race.models;

import android.content.ContentValues;

import ua.hospes.nfs.marathon.core.db.ModelBaseInterface;
import ua.hospes.nfs.marathon.core.db.tables.Race;

/**
 * @author Andrew Khloponin
 */
public class RaceItemDb implements ModelBaseInterface {
    private int id = -1;
    private final int teamId;
    private int teamNumber = -1;
    private int sessionId = -1;
    private int order = 0;


    public RaceItemDb(int teamId) {
        this.teamId = teamId;
    }


    //region Getters
    public int getId() {
        return id;
    }

    public int getTeamId() {
        return teamId;
    }

    public int getTeamNumber() {
        return teamNumber;
    }

    public int getSessionId() {
        return sessionId;
    }

    public int getOrder() {
        return order;
    }
    //endregion

    //region Setters
    public void setId(int id) {
        this.id = id;
    }

    public void setTeamNumber(int teamNumber) {
        this.teamNumber = teamNumber;
    }

    public void setSessionId(int sessionId) {
        this.sessionId = sessionId;
    }

    public void setOrder(int order) {
        this.order = order;
    }
    //endregion


    @Override
    public String toString() {
        return "RaceItemDb{" +
                "id=" + id +
                ", teamId=" + teamId +
                ", teamNumber=" + teamNumber +
                ", sessionId=" + sessionId +
                ", order=" + order +
                '}';
    }

    @Override
    public ContentValues toContentValues() {
        ContentValues cv = new ContentValues();

        cv.put(Race.TEAM_ID, teamId);
        cv.put(Race.TEAM_NUMBER, teamNumber);
        cv.put(Race.SESSION_ID, sessionId);

        return cv;
    }
}

File: app/src/main/java/ua/hospes/rtm/core/db/race/models/RaceItemDb.kt
package ua.hospes.rtm.core.db.race.models

import android.content.ContentValues

import ua.hospes.dbhelper.IDbModel
import ua.hospes.rtm.core.db.tables.Race

/**
 * @author Andrew Khloponin
 */
class RaceItemDb(val teamId: Int) : IDbModel {
    //region Getters
    //endregion

    //region Setters
    var id = -1
    var teamNumber = -1
    var sessionId = -1
    var order = 0
    //endregion


    override fun toString(): String {
        return "RaceItemDb{" +
                "id=" + id +
                ", teamId=" + teamId +
                ", teamNumber=" + teamNumber +
                ", sessionId=" + sessionId +
                ", order=" + order +
                '}'.toString()
    }

    override fun toContentValues(): ContentValues {
        val cv = ContentValues()

        cv.put(Race.TEAM_ID.name(), teamId)
        cv.put(Race.TEAM_NUMBER.name(), teamNumber)
        cv.put(Race.SESSION_ID.name(), sessionId)

        return cv
    }
}